components/page-templates/B2B/UsersTemplate/UsersTemplate.tsx

```diff
@@ -1,18 +1,309 @@
+    <Button
+      variant="primary"
+      disabled={isUserFormOpen}
+      onClick={onClick}
+      disableElevation
+      id="formOpenButton"
+      startIcon={<AddCircleOutlineIcon />}
+      sx={sx}
+    >
+      <span>{t('add-user')}</span>
+    </Button>
+  )
+}
+
+const UsersTemplate = () => {
+  const {
+    publicRuntimeConfig: {
+      b2bUserRoles,
+      debounceTimeout,
+      b2bUserListing: { defaultPageSize, defaultStartIndex, defaultFilter },
+    },
+  } = getConfig()
+  const userRoles = b2bUserRoles
+
+  const theme = useTheme()
+  const { user } = useAuthContext()
+  const { t } = useTranslation('common')
+  const { showModal, closeModal } = useModalContext()
+  const mdScreen = useMediaQuery(theme.breakpoints.up('md'))
+
+  const [isUserFormOpen, setIsUserFormOpen] = useState<boolean>(false)
+
+  const [paginationState, setPaginationState] = useState({
+    searchTerm: '',
+    pageSize: defaultPageSize,
+    startIndex: defaultStartIndex,
+  })
 
-        Add your code here
+  const { data, isLoading } = useGetB2BUserQueries({
+    accountId: user?.id as number,
+    filter: defaultFilter,
+    pageSize: paginationState.pageSize,
+    startIndex: paginationState.startIndex,
+    q: useDebounce(paginationState.searchTerm, debounceTimeout),
+  })
+
+  const { removeCustomerB2bUser } = useRemoveCustomerB2bUserMutation()
+  const { createCustomerB2bUser } = useCreateCustomerB2bUserMutation()
+  const { addRoleToCustomerB2bAccount } = useAddRoleToCustomerB2bAccountMutation()
+  const { updateCustomerB2bUser } = useUpdateCustomerB2bUserMutation()
+  const { deleteB2bAccountUserRole } = useDeleteB2bAccountRoleMutation()
+
+  const confirmDelete = (id: string | undefined | null) => {
+    showModal({
+      Component: ConfirmationDialog,
+      props: {
+        contentText: t('delete-user-confirmation-text'),
+        primaryButtonText: t('yes-remove'),
+        title: t('confirmation'),
+        onConfirm: () => {
+          const accountId = user?.id
+          const queryVars = { accountId, userId: id }
+          removeCustomerB2bUser.mutateAsync({ ...queryVars })
+        },
+      },
+    })
+  }
+
+  const handleSearch = (searchText: string) => {
+    setPaginationState({
+      ...paginationState,
+      searchTerm: searchText,
+      startIndex: defaultStartIndex,
+    })
+  }
+
+  const handlePageChange = (event: ChangeEvent<any>, page: number) =>
+    setPaginationState({
+      ...paginationState,
+      startIndex: (data?.pageSize ?? 0) * (page - 1),
+    })
+
+  const handleAddUser = async (formValues: B2BUserInput) => {
+    const variables = buildCreateCustomerB2bUserParams({ user, values: formValues })
+    const createUserResponse = await createCustomerB2bUser.mutateAsync({
+      ...variables,
+    })
+    if (createUserResponse?.userId) {
+      addRoleToB2bUser(createUserResponse, formValues)
+    }
+  }
+
+  const handleUpdateUser = async (formValues: B2BUserInput, b2BUser?: B2BUser | undefined) => {
```

Comment:
b2BUser?: B2BUser | undefined

If we are using ?, no need to explicitly provide undefined again

Reply:
Done